From e70b39ab235c0ef7390ea8cec91f05428d2e775a Mon Sep 17 00:00:00 2001
From: Andrey Borzenkov <arvidjaar@gmail.com>
Date: Mon, 25 Feb 2013 22:53:40 +0100
Subject: [PATCH 159/482] 	* grub-core/normal/menu_entry.c
 (update_screen): remove 	unused variable `off' which caused scroll down
 arrow to be always shown.

---
 ChangeLog                     | 5 +++++
 grub-core/normal/menu_entry.c | 3 +--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index cc5d5e3..1b66689 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-02-25  Andrey Borzenkov <arvidjaar@gmail.com>
 
+	* grub-core/normal/menu_entry.c (update_screen): remove
+	unused variable `off' which caused scroll down arrow to be always shown.
+
+2013-02-25  Andrey Borzenkov <arvidjaar@gmail.com>
+
 	* grub-core/normal/menu_entry.c (insert_string): fix off by one
 	access to unallocated memory.
 
diff --git a/grub-core/normal/menu_entry.c b/grub-core/normal/menu_entry.c
index 85f97da..f4c8afd 100644
--- a/grub-core/normal/menu_entry.c
+++ b/grub-core/normal/menu_entry.c
@@ -243,7 +243,6 @@ update_screen (struct screen *screen, struct per_term_screen *term_screen,
 
       do
 	{
-	  int off = 0;
 	  struct grub_term_pos **pos;
 
 	  if (linep >= screen->lines + screen->num_lines)
@@ -301,7 +300,7 @@ update_screen (struct screen *screen, struct per_term_screen *term_screen,
 	  y += get_logical_num_lines (linep, term_screen);
 	  if (y >= term_screen->num_entries)
 	    {
-	      if (off <= linep->len || i + 1 < screen->num_lines)
+	      if (i + 1 < screen->num_lines)
 		down_flag = 1;
 	    }
 
-- 
1.8.2.1

